File: frontend/src/services/roleService.js
import api from './api';

export const roleService = {
  getRoles: (params) => api.get('/roles', { params }),
  getRoleById: (id) => api.get(`/roles/${id}`),
  createRole: (data) => api.post('/roles', data),
  updateRole: (id, data) => api.put(`/roles/${id}`, data),
  deleteRole: (id) => api.delete(`/roles/${id}`),
};

export default roleService;

File: backend/src/main/java/com/workpulse/ems/service/RoleService.java
package com.workpulse.ems.service;

import com.workpulse.ems.dto.request.RoleRequest;
import com.workpulse.ems.dto.response.RoleResponse;

import java.util.List;

public interface RoleService {

    List<RoleResponse> getAllRoles();

    RoleResponse getRoleById(Long id);

    RoleResponse createRole(RoleRequest request);

    RoleResponse updateRole(Long id, RoleRequest request);

    void deleteRole(Long id);
}
